test/jdk/sun/security/validator/CertReplace.java

```diff
@@ -21,27 +21,186 @@
+                        certInputStream
+                )
+        };
+        certInputStream.close();
+
+        final PrivateKey privateKey = (PrivateKey) keyStore.getKey("int", PASSWORD_CHAR_ARR);
+        keyStore.setKeyEntry("int", privateKey, PASSWORD_CHAR_ARR, certs);
+        keyStore.store(new FileOutputStream(CERTREPLACE_JKS), PASSWORD_CHAR_ARR);
+
+
+        SecurityTools.keytool(ktBaseParameters +
+                              "-certreq -alias user -file user.req");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-gencert -rfc -alias int " +
+                              "-infile user.req " +
+                              "-outfile certreplace.certs"); // this will create certreplace.certs which is later appended
+
+        // 3. Create the certchain file
+        final Path certPath = Paths.get("certreplace.certs");
+
+        Files.write(certPath, Files.readAllBytes(Path.of("int.cert")), StandardOpenOption.APPEND);
+
+        final String outputCa = SecurityTools.keytool(ktBaseParameters +
+                                                      "-export -rfc -alias ca").getOutput();
+        Files.write(certPath, outputCa.getBytes(), StandardOpenOption.APPEND);
+
+        // 4. Upgrade ca from MD2withRSA to SHA256withRSA, remove other aliases and make this keystore the cacerts file
+        keyStore.deleteEntry("int");
+        keyStore.deleteEntry("user");
+        keyStore.store(new FileOutputStream(CERTREPLACE_JKS), PASSWORD_CHAR_ARR);
+
+        SecurityTools.keytool(ktBaseParameters +
+                              "-selfcert -alias ca");
+    }
+
+    /**
+     * This method creates certs for the Same DN test
+     *
+     * @throws Exception
+     */
+    private static void sameDn() throws Exception {
+
+        final String ktBaseParameters = "-storepass " + PASSWORD + " " +
+                                        "-keypass " + PASSWORD + " " +
+                                        "-keystore " + SAMEDN_JKS + " " +
+                                        "-keyalg rsa ";
+
+        final Path keystoreFilePath = Paths.get(SAMEDN_JKS);
+        Files.deleteIfExists(keystoreFilePath);
+
+        // 1. Generate 3 aliases in a keystore: ca1, ca2, user. The CAs' startdate
+        // is set to one year ago so that they are expired now
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias ca1 -dname CN=CA -keyalg rsa " +
+                              "-sigalg md5withrsa -ext bc -startdate -1y");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias ca2 -dname CN=CA -keyalg rsa " +
+                              "-sigalg sha1withrsa -ext bc -startdate -1y");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias user -dname CN=User -keyalg rsa");
+
+        final KeyStore keyStore = KeyStoreUtils.loadKeyStore(SAMEDN_JKS, PASSWORD);
```

Comment:
Although no change to behavior in this method, I'd prefer seeing this line right before the `deleteEntry` call.